### PR TITLE
Try to upgrade tinkerpop version to 3.5.1 in maxgraph

### PR DESCRIPTION
**Pull request by shirly121**
## What do these changes do?
1, upgrade tinkerpop under the interactive_engine from 3.4.12 to 3.5.1.
2, update codes of the compiler (used by groot and maxgraph) for some changes of interface in 3.5.1.
3, update unit tests of the compiler for the changes in 3.5.1.
4, update ci of groot and maxgraph by removing 7 blocking test cases.
5, ignore gaia tests in the ci procedure for we will replace it by ir.

## Related issue number

Fixes

### CODE REVIEW

**Comment by longbinlai** [comment]
Could you please add more information for what this PR does? What has been done and why?

File: .github/workflows/gaia.yml [hunk 1]
```diff
@@ -1,70 +0,0 @@
-name: GraphScope GAIA CI
```
Comment (yecol) [inline .github/workflows/gaia.yml:1]: If you intend to disable this CI temporarily, it would be better to comment out rather than remove this file.

**Review by yecol — CHANGES_REQUESTED** [review]
Please double check the changes on graphlearn.
It seems this PR should not introduce changes to it?

File: .github/workflows/gaia.yml (hunk 1 above)
Comment (sighingnow) [reply .github/workflows/gaia.yml:1]: The GAIA ci has already been disabled before (see line 23) and not used anymore, as the related codepath has already been dropped.

Comment (shirly121) [reply .github/workflows/gaia.yml]: okay, I have recovered the gaia.yaml

**Comment by shirly121** [comment]
> Could you please add more information for what this PR does? What has been done and why?

done

**Comment by longbinlai** [comment]
Please resolve the conflict @shirly121

**Comment by shirly121** [comment]
>

**Comment by shirly121** [comment]
> Please double check the changes on graphlearn. It seems this PR should not introduce changes to it?

It's a mistake, I have reverted the changes.

**Review by longbinlai — APPROVED** [review]
LGTM

**Review by yecol — APPROVED** [review]

**Pull request merged**